specification/schemaregistry/data-plane/Microsoft.EventHub/stable/2022-10/schemaregistry.json

```diff
@@ -0,0 +1,615 @@
+            "name": "groupName",
+            "in": "path",
+            "description": "Schema group under which schema is registered.  Group's serialization type should match the serialization type specified in the request.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/SchemaName"
+          },
+          {
+            "$ref": "#/parameters/SchemaVersion"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "headers": {
+              "Location": {
+                "type": "string",
+                "description": "URL location of schema, identified by schema group, schema name, and version."
+              },
+              "Content-Type": {
+                "description": "The content type for given schema. Each schema type has an associated content-type.",
+                "type": "string"
+              },
+              "Schema-Id": {
+                "type": "string",
+                "description": "References specific schema in registry namespace."
+              },
+              "Schema-Id-Location": {
+                "type": "string",
+                "description": "URL location of schema, identified by schema ID."
+              },
+              "Schema-Group-Name": {
+                "type": "string",
+                "description": "References schema group."
+              },
+              "Schema-Name": {
+                "type": "string",
+                "description": "References schema name."
+              },
+              "Schema-Version": {
+                "type": "integer",
+                "format": "int32",
+                "description": "Version of the returned schema."
+              }
+            },
+            "schema": {
+              "description": "Binary representation of the registered schema.",
+              "type": "file"
+            }
+          },
+          "default": {
+            "description": "An error response received from the Azure Schema Registry service.",
+            "headers": {
+              "x-ms-error-code": {
+                "type": "string",
+                "description": "Error code for specific error that occurred."
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        }
+      }
+    },
+    "/$schemaGroups/{groupName}/schemas/{schemaName}:get-id": {
+      "post": {
+        "summary": "Get ID for existing schema.",
+        "description": "Gets the ID referencing an existing schema within the specified schema group, as matched by schema content comparison.",
+        "x-ms-examples": {
+          "Get schema ID": {
+            "$ref": "./examples/OperationSchema_QueryIdByContent.json"
+          }
+        },
+        "operationId": "Schema_QueryIdByContent",
```

Comment:
This is a misleading operation. You're using a POST for what effectively could be a GET e.g. "/$schemaGroups/{groupName}/schemas/{schemaName}" that returns a schema with an `id` property. Why make this a POST in lieu of a GET?

Reply:
Good point, perhaps to remedy this, the service could also support GET requests in another but similar endpoint?

Reply:
We made this a POST request because this method returns the schema id for the schema that is passed in the body of the request, and we currently don't have a way of handling GET requests that include a request body.

Reply:
Yes, I don't think GET requests are meant to contain a body. From the [spec](https://www.rfc-editor.org/rfc/rfc7231#section-4.3.1):

> A payload within a GET request message has no defined semantics;
   sending a payload body on a GET request might cause some existing
   implementations to reject the request.

Reply:
Oh right, I missed the included body part.